thingif/src/main/java/com/kii/thing_if/gateway/GatewayCommandAPI.java

```diff
@@ -0,0 +1,213 @@
+     * @param app app instance.
+     * @param owner owner of gateway.
+     * @param gatewayID thingID of gateway.
+     */
+    public GatewayCommandAPI(
+            @NonNull KiiApp app,
+            @NonNull Owner owner,
+            @NonNull String gatewayID) {
+        this.app = app;
+        this.owner = owner;
+        this.gatewayID = gatewayID;
+        this.restClient = new IoTRestClient();
+    }
+
+    /**
+     * Send command to gateway to get list of pending nodes in gateway.
+     * @return command id
+     * @throws ThingIFException
+     */
+    public String postListPendingEndNodesCommand() throws ThingIFException {
+        JSONObject requestBody = new JSONObject();
+        try {
+            requestBody.put("issuer", this.owner.getTypedID().toString());
+            requestBody.put("actions", new JSONArray()
+                    .put(new JSONObject()
+                            .put("listPendings", new JSONObject()
+                                    .put("serverLocation", this.app.getSiteName())
+                                    .put("appID", this.app.getAppID())
+                            )
+                    )
+            );
+        }catch (JSONException ex) {
+            // wont happend
+        }
+        return postNewCommand(requestBody);
+    }
+
+    /**
+     * Send command to gateway to notify it with completion of onboarding for end node.
+     * @param endNode EndNode instance.
+     * @return command id
+     * @throws ThingIFException
+     */
+    public String postOnboardCompletionCommand(@NonNull EndNode endNode) throws ThingIFException {
+        JSONObject requestBody = new JSONObject();
+        try {
+            requestBody.put("issuer", this.owner.getTypedID().toString());
+            requestBody.put("actions", new JSONArray()
+                    .put(new JSONObject()
+                            .put("onboardEndnode", new JSONObject()
+                                    .put("serverLocation", this.app.getSiteName())
+                                    .put("appID", this.app.getAppID())
+                                    .put("vendorThingID", endNode.getVendorThingID())
+                                    .put("thingID", endNode.getThingID())
+                            )
+                    )
+            );
+        }catch (JSONException ex) {
+            // wont happend
+        }
+        return postNewCommand(requestBody);
+    }
+
+    @NonNull
+    public KiiApp getApp() {
+        return app;
+    }
+
+    @NonNull
+    public Owner getOwner() {
+        return owner;
+    }
+
+    @NonNull
+    public String getGatewayID() {
+        return gatewayID;
+    }
+
+    /**
+     * Get action results of list pending endnode command.
+     * @param commandID command id.
+     * @return ActionResult instance.
+     * @throws ThingIFException
+     */
+    public ActionResult getListPendingEndNodesResult(String commandID) throws ThingIFException {
```

Comment:
How to distinguish the execution failure and the case not yet responded by the target?